File: src/com/blackoutburst/game/EntityOtherPlayer.java
package com.blackoutburst.game;

import com.blackoutburst.bogel.graphics.Color;
import com.blackoutburst.bogel.maths.Vector2f;
import com.blackoutburst.bogel.maths.Vector3f;

public class EntityOtherPlayer extends Entity {


	public EntityOtherPlayer(int id, EntityTypes type, Vector3f position, Vector3f scale, Vector3f rotation) {
		super(id, type, position, scale, rotation);
	}

	@Override
	protected void update() {}

	@Override
	protected void render() {
		new Cube(Textures.MUSH, position, scale, rotation, Color.WHITE, new Vector2f(), false).draw();
	}
}

File: client/src/main/kotlin/dev/blackoutburst/game/core/entity/EntityOtherPlayer.kt
package dev.blackoutburst.game.core.entity

import dev.blackoutburst.game.graphics.Color
import dev.blackoutburst.game.graphics.Cube
import dev.blackoutburst.game.maths.Vector2f
import dev.blackoutburst.game.maths.Vector3f

class EntityOtherPlayer(
    id: Int,
    position: Vector3f = Vector3f(),
    rotation: Vector2f = Vector2f(),
) : Entity(id, position, rotation) {

    private val cube = Cube(position, rotation, Color(0.8f, 0.2f, 0.6f))

    override fun update() {
        cube.position = position
        cube.rotation = rotation
    }

    override fun render() {
        cube.draw()
    }

}